### ISSUE TITLE
GetList throws an exception when ExpressionFilter, Fields, and SortedFields aren't provided

### ISSUE THREAD

**Issue by craibuc** [intacct/intacct-sdk-net#183]
I'm attempting to use this assembly with PowerShell:

```powershell
# Intacct.ps1

$Here = Split-Path -Parent $MyInvocation.MyCommand.Path
Push-Location $Here

Add-Type -AssemblyName 'Intacct.SDK.dll'

$UserCredential = [pscredential]::new('USER', ('PASSWORD' | ConvertTo-SecureString -AsPlainText))
$SenderCredential = [pscredential]::new('SENDER', ('PASSWORD' | ConvertTo-SecureString -AsPlainText))
$CompanyId = 'COMPANY'

# $Field = 'name'

# $SortedField = @{
#     Name='name'
#     Order='asc'
# }

# $Expression = @{
#     Field = "name"
#     Operator = "="
#     Value = 'Bills'
# }

try
{
    [Intacct.SDK.Functions.ApiSessionCreate]$ApiSessionCreate = [Intacct.SDK.Functions.ApiSessionCreate]::new()

    [Intacct.SDK.ClientConfig]$ClientConfig = [Intacct.SDK.ClientConfig]::new()
    $ClientConfig.SenderId = $SenderCredential.UserName
    $ClientConfig.SenderPassword = $SenderCredential.Password | ConvertFrom-SecureString -AsPlainText
    $ClientConfig.UserId = $UserCredential.UserName
    $ClientConfig.UserPassword = $UserCredential.Password | ConvertFrom-SecureString -AsPlainText
    $ClientConfig.CompanyId = $CompanyId

    [Intacct.SDK.OnlineClient]$OnlineClient = [Intacct.SDK.OnlineClient]::new($ClientConfig)

    if ( $ClientConfig ) { [Intacct.SDK.OnlineClient]$OnlineClient = [Intacct.SDK.OnlineClient]::new($ClientConfig) }
    else { throw "Credentials not found" }

    [Intacct.SDK.Functions.Common.List.GetList]$List = [Intacct.SDK.Functions.Common.List.GetList]::new('supdocfolder')
    $List.maxitems = 100

    if ( $null -ne $Expression ) { $List.Expression = [Intacct.SDK.Functions.Common.List.ExpressionFilter]::new($Expression.Field, $Expression.Operator, $Expression.Value) }

    if ( $null -ne $Field ) { $List.Fields = $Field }
    
    foreach ($sort in $SortedField)
    {
        $List.SortedFields.Add( [Intacct.SDK.Functions.Common.List.SortedField]::new($sort.Name, $sort.Order) )
    }
    
    do 
    {
        [System.Threading.Tasks.Task[Intacct.SDK.Xml.OnlineResponse]]$task = $OnlineClient.Execute($List)
        $task.Wait()
    
        # results of query
        [Intacct.SDK.Xml.OnlineResponse]$Response = $task.Result
    
        # return the Data node
        Write-Output ( [Newtonsoft.Json.JsonConvert]::SerializeObject( $Response.Results[0].Data ) | ConvertFrom-Json ).supdocfolder

        # redefine $Reader to be a ReadMore, instead of a ReadByQuery
        [Intacct.SDK.Functions.Common.ReadMore]$Reader = [Intacct.SDK.Functions.Common.ReadMore]::new()
        $Reader.ResultId = $Response.Results[0].ResultId

    } while ($Response.Results[0].NumRemaining -gt 0)
}
catch 
{
    Write-Host $_.Exception.Message -ForegroundColor Red
}
```

When the execution reaches the `$task.Wait()` line, an exception is thrown:

```powershell
Exception calling "Wait" with "0" argument(s): "One or more errors occurred. (Response control status failure - XL03000003 XML Parse schema error: Error 1871: Element 'filter': Missing child element(s). Expected is one of ( logical, expression ).. Line: 1, column: 0. [Support ID: 7kDfyEB034%7EYYrdsP0v2lT-nTBW%40Uy8%40gAAAAE] - XL03000003 XML Parse schema error: Error 1871: Element 'sorts': Missing child element(s). Expected is ( sortfield ).. Line: 1, column: 0. - XL03000003 XML Parse schema error: Error 1871: Element 'fields': Missing child element(s). Expected is ( field ).. Line: 1, column: 0.)"
```

When values are supplied for `$Field`, `$SortedField`, and `$Expression`, no exception is raised.

If the XML API is used, the `filter`, `sorts`, and `fields` nodes are optional:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<request>
  <control>
    <senderid>{{sender_id}}</senderid>
    <password>{{sender_password}}</password>
    <controlid>{{$timestamp}}</controlid>
    <uniqueid>false</uniqueid>
    <dtdversion>3.0</dtdversion>
    <includewhitespace>false</includewhitespace>
  </control>
  <operation>
    <authentication>
      <sessionid>{{temp_session_id}}</sessionid>
    </authentication>
    <content>
      <function controlid="{{$guid}}">
        <get_list object="supdocfolder" maxitems="100">
            <!-- <filter><expression><field>name</field><operator>=</operator><value>Lorenz</value></expression></filter> -->
            <!-- <sorts><sortfield order="desc">name</sortfield></sorts> -->
            <!-- <fields><field>name</field></fields> -->
        </get_list>
      </function>
    </content>
  </operation>
</request>
```
Why does the .Net client appear to need these properties that the XML API does not?

To reproduce:

Save script as `Intacct.ps1`

Ensure these assemblies are in the same directory as the script:
* Intacct.SDK.dll (latest version)
* Microsoft.Extensions.FileProviders.Abstractions 2.dll
* Microsoft.Extensions.FileProviders.Abstractions.dll
* Microsoft.Extensions.FileProviders.Physical.dll
* Microsoft.Extensions.Logging.Abstractions.dll
* Microsoft.Extensions.Primitives.dll
* Newtonsoft.Json.dll

Run the script in PowerShell v7.

**Comment by craibuc** [comment]
Issues in the code (https://github.com/Intacct/intacct-sdk-net/blob/master/Intacct.SDK/Functions/Common/List/GetList.cs):

* Lines 66 adds `<filter>` and line 93 adds `</filter>`, which is illegal without child nodes.
* Lines 95 adds `<sorts>` and line 106 adds `</sorts>`, which is illegal without child nodes.
* Lines 108 adds `<fields>` and line 117 adds `</fields>`, which is illegal without child nodes.

The code should be:

```
if (Expression != null || Logical != null)
{
  xml.WriteStartElement("filter");
  ...
  xml.WriteEndElement(); // </filter>
}

if (this.SortedFields.Count > 0)
{
  xml.WriteStartElement("sorts");
  ...
  xml.WriteEndElement(); // </sorts>
}

if (this.Fields.Count > 0)
{
  xml.WriteStartElement("fields");
  ...
  xml.WriteEndElement(); // </fields>
}
```

**Comment by dylan-holt-sage** [comment]
@craibuc, thanks for your PR, I have reviewed and merged the code in our upcoming version 3.2.0, release TBD.
